src/zenml/integrations/s3/flavors/s3_artifact_store_flavor.py

```diff
@@ -106,6 +113,40 @@ def _validate_client_kwargs(
             )
         return value
 
+    @property
+    def bucket(self) -> str:
+        """The bucket name of the artifact store.
+
+        Returns:
+            The bucket name of the artifact store.
+        """
+        if self._bucket is None:
+            self._bucket = os.path.normpath(self.path).split(os.sep)[1]
```

Comment:
Will this work on windows, where `os.sep` is not `/`?

Reply:
Ouch, indeed